sys/fmt/fmt.c

```diff
@@ -62,6 +62,19 @@ size_t fmt_byte_hex(char *out, uint8_t byte)
     return 2;
 }
 
+size_t fmt_bytes_hex(char *out, const uint8_t *ptr, size_t n)
+{
+    size_t len = 0;
+    if (out) {
+        while (n--) {
+            out += fmt_byte_hex(out, *ptr++);
+            len += 2;
```

Comment:
This way len is broken if there's no ```out```.

This should fix it: https://gist.github.com/06d9b04ed210aed79d42bd4bba885298

Reply:
Maybe add a unittest for ```out==NULL``` but ```n > 0```.

Reply:
done